### PR TITLE
Table Accessibility

### PR DESCRIPTION
**Pull request by okkays**
### Adds an expanded/collapsed indicator
- ⛔ The report list table doesn't include state (expanded/collapsed) or
  role (button, not link) properly, and the wording is off.
- ✅ This commit improves them by:
  * Tweaking some of the wording
  * Using aria-expanded and role to indicate properly that it's an
    expand/collapse button.

### Adds more accurate column headers
- 🌎 Screen readers read the column header when navigating between table
  cells to indicate which column you're on, e.g. "View Count. 15."
- ⛔ The report list were reads as "Sort by x ascending" for the column header.
- ✅ This commit changes the column label to the heading (Status, etc).
  * When readers are on the headers, they will say, "Status. Sort by status"
  * When readers are on the cells, they will say, "Status. Link. New."

### Improves checkbox/expansion wording
- 🌎 This commit relates to the accessibility of the /form/view table. Screen readers current read the checkbox header as "checkboxes".
- ⛔ That doesn't give any info about what the checkboxes do. Smart
  screen readers default to "Select all records" which isn't great,
  either, especially when it reads it on individual rows.
- ✅ This commit adds a header to the checkboxes and to the dropdowns to
  make it a bit more descriptive.

### Improves detail row interaction
- ⛔ Screen readers don't read the detail rows properly
- ✅ This commit helps make them more accessible by providing
  appropriate table headers and aria roles.

## Screenshots (for front-end PR):

A quick demo of this feature can be found here (note - this demo excludes the redundant headers found based on aXe failures):

https://user-images.githubusercontent.com/15126660/212344533-8ce50b76-2abe-4d82-abd4-d7b493968b1a.mp4

### Author

+ [x] If this is a story, run locally and check to make sure all Acceptance Criteria are met. (If any criteria are unclear, ask about them.).
+ [x] Check for, document, and establish a testing plan for any behavior that may vary across environments or is otherwise difficult to test.
+ [x] Check for [accessibility](https://github.com/usdoj-crt/crt-portal/blob/develop/docs/a11y_plan.md).
+ [x] [Tests pass](https://github.com/USDOJ/crt-portal/#tests).

### Reviewer

+ [ ] If this is a story, run locally and check to make sure all Acceptance Criteria are met. (If any criteria are unclear, ask about them.).
+ [ ] Check for any behavior that may vary across environments or is difficult to test, and ensure that it is well-understood, documented, and that there is a testing plan in place.
+ [ ] Re-check for [accessibility](https://github.com/usdoj-crt/crt-portal/blob/develop/docs/a11y_plan.md).
+ [ ] [Tests pass](https://github.com/USDOJ/crt-portal/#tests).

## Notes for reviewer:

See [PR instructions doc](https://github.com/usdoj/crt-portal/blob/master/docs/pull_requests.md) for full pull request review instructions.

### CODE REVIEW

File: crt_portal/cts_forms/templates/forms/snippets/sortable_table_heading.html
Comment (JackRyan1989) [inline crt_portal/cts_forms/templates/forms/snippets/sortable_table_heading.html:13]: Ok, so there's some weird stuff that happens here, really only for the headings without `<a>` tags as children. If it's just an aria-hidden span element, the heading never receives focus when tabbing through. Which is really problematic for keyboard users. Even when using a screen reader the element still doesn't receive visual focus. I would suggest wrapping the aria hidden span elements in something like a tab-indexed `<div>` element and see if that adds back keyboard focus.

Also, NVDA doesn't read these in Edge. It skips over Contact Details and Incident City/State headings.

**Review by JackRyan1989 — CHANGES_REQUESTED** [review]
I have some requested changes. I think this is super close, but there are some weird things happening, mostly related to tab index and hiding certain elements.

File: crt_portal/cts_forms/templates/forms/complaint_view/index/complaints_table.html
Comment (okkays) [reply crt_portal/cts_forms/templates/forms/complaint_view/index/complaints_table.html:62]: That would display the text, e.g.:

image

We could add the text and mark as usa-sr-only, but that actually hides the checkbox as well:

image

File: crt_portal/cts_forms/templates/forms/complaint_view/index/complaints_table.html
Comment (okkays) [reply crt_portal/cts_forms/templates/forms/complaint_view/index/complaints_table.html:123]: Because this is inside of a `<td>`, it's redundant for screen readers, as they already read the column header. It'd be like having this for other values:

image

Plus, screen readers jump to that header span as a target before jumping to the value, which is inconsistent with how the rest of the table cells work.

That said, the setup is still not ideal - because of the documented lack of support for `headers` in some screen readers, there _is_ an `aria-label` on the td as a backup to avoid it reading:

`Submitted. This is a personal description...`

instead of 

`Personal Description. This is a personal description...`

So our final behavior is:

`(Submitted | Personal Description). Personal Description. This is a personal description...`

File: crt_portal/cts_forms/templates/forms/complaint_view/index/complaints_table.html
Comment (JackRyan1989) [reply crt_portal/cts_forms/templates/forms/complaint_view/index/complaints_table.html:62]: Ok cool. Thanks!

File: crt_portal/cts_forms/templates/forms/complaint_view/index/complaints_table.html
Comment (ilodidoj) [reply crt_portal/cts_forms/templates/forms/complaint_view/index/complaints_table.html:13]: @okkays @JackRyan1989 Just FYI, portal internal site only be tested against internal approved browsers, i.e. Microsoft EDGE and Google Chrome latest or one minor version behind from latest browser version.

File: crt_portal/cts_forms/templates/forms/complaint_view/index/complaints_table.html
Comment (okkays) [reply crt_portal/cts_forms/templates/forms/complaint_view/index/complaints_table.html:121]: > It's very easy to blow past the summary row that was just opened

This makes sense to me for the expansion toggle - I've updated it so going from the first or second column down will enter the details row

For the details box, I don't think we can, or maybe we don't even want to, jump to it from other columns based on alignment. If someone has some rows expanded, but wants to scan through ID, for example, they might prefer "ID: 5059, ID: 5060, ID: 5061" vs "ID: 5059, Summary, ID: 5060, Summary, ...". Beyond that UX, technically, I don't see a way to tell it to jump from a group of the cells above to the colspan cell below.

Going from submitted down will jump into Details, but I think that's unavoidable so long as this exists as a table row, and VoiceOver jumps right over it if I try to set it up as not a row due to the way it navigates between table rows.

File: crt_portal/cts_forms/templates/forms/snippets/sortable_table_heading.html
Comment (okkays) [reply crt_portal/cts_forms/templates/forms/snippets/sortable_table_heading.html:13]: Nice, good catch - adding tabindex seems to have added back tab/voiceover focus for the headers and cells

File: crt_portal/cts_forms/templates/forms/complaint_view/index/complaints_table.html
Comment (okkays) [reply crt_portal/cts_forms/templates/forms/complaint_view/index/complaints_table.html:13]: I've added a tabindex of -1 here to try to get the screen readers to ignore it.

Unfortunately, hiding this table or moving it elsewhere causes readers to miss it when looking for the matching `headers` attribute.

**Review by ilodidoj — COMMENTED** [review]
@okkays if @JackRyan1989 OK with your comments response, you can merge this to dev after his approval. Thanks, nice work!!

**Comment by ilodidoj** [comment]
@okkays @JackRyan1989 I will create a release branch from dev today once this is merged to dev. Thanks.

File: crt_portal/cts_forms/templates/forms/complaint_view/index/complaints_table.html
Comment (JackRyan1989) [reply crt_portal/cts_forms/templates/forms/complaint_view/index/complaints_table.html:13]: I think you can remove the change here @okkays 

The `tabindex=-1` doesn't fix the issue on Firefox, and from @ilodidoj's comment it's a moot point anyway. It works fine on Chrome so I think we're ok.

File: crt_portal/cts_forms/templates/forms/complaint_view/index/complaints_table.html
Comment (JackRyan1989) [reply crt_portal/cts_forms/templates/forms/complaint_view/index/complaints_table.html:124]: So I think you're right about the details box. One thing to note though is that VoiceOver reads different headings depending on the direction you're moving through the table. This happens in Chrome, not Firefox. 

What happens is if you navigate into Personal Description cell from a cell above it the heading of the cell is read correctly. However, if you try to navigate from a cell adjacent to it, the table heading for that column is read. Not sure how we can fix this, since it might be down to how the browsers handle the headers attribute.

**Review by JackRyan1989 — APPROVED** [review]
I think this looks good. I think it's ready to move onto Terri's review.

File: crt_portal/cts_forms/templates/forms/complaint_view/index/complaints_table.html
Comment (okkays) [reply crt_portal/cts_forms/templates/forms/complaint_view/index/complaints_table.html:124]: Yep - iiuc it's not something we can fix. It's what the `headers` attribute was designed for, but not all readers support it yet. I've added a bit of redundancy in the meantime so it will at least read the correct heading once.

**Pull request merged**
